### PR TITLE
[lmdb] don't use msvc parameters with non-msvc compiler

### PR DESCRIPTION
**Pull request by bold84**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes failed build on non-windows-dynamic

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/lmdb/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "lmdb",
-  "version-string": "0.9.24",
+  "version": "0.9.24",
   "port-version": 1,
```
Comment (Thomas1664) [inline ports/lmdb/vcpkg.json:4]:
```suggestion
  "port-version": 2,
```

File: versions/l-/lmdb.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "versions": [
     {
-      "git-tree": "f3599ec6a6285945a3eeeac1f9b0db810855ec5a",
-      "version-string": "0.9.24",
+      "git-tree": "5ed5bd2c01a6c4467f9be85807e7727666ddb5b7",
+      "version": "0.9.24",
```
Comment (Thomas1664) [inline versions/l-/lmdb.json:5]: please revert this and run `vcpkg x-add-version lmdb` again after you committed this and the other changes.

File: ports/lmdb/cmake/CMakeLists.txt [hunk 3]
```diff
@@ -20,8 +20,8 @@ set(LMDB_LIBRARY_INSTALL_DIR lib CACHE PATH "Install directory for library")
 set(LMDB_RUNTIME_INSTALL_DIR bin CACHE PATH "Install directory for binaries/dlls")
 set(LMDB_CONFIG_INSTALL_DIR share/lmdb CACHE PATH "Install directory for cmake config files")
 
-
-if(BUILD_SHARED_LIBS)
+message("Compiler: ${CMAKE_CXX_COMPILER_ID}")
```
Comment (Thomas1664) [inline ports/lmdb/cmake/CMakeLists.txt:23]: Why is this message needed?

**Review by Thomas1664 — CHANGES_REQUESTED** [review]

File: ports/lmdb/cmake/CMakeLists.txt (hunk 3 above)
Comment (bold84) [reply ports/lmdb/cmake/CMakeLists.txt:23]: Nope. Forgot to remove it...

**Comment by JonLiu1993** [comment]
@bold84 ,Thanks for your pr, please consider @Thomas1664's review suggesstion

**Comment by bold84** [comment]
> @bold84 ,Thanks for your pr, please consider @Thomas1664's review suggesstion

Sure, it's just a pita that for a one line code change, it is required to do these version things that I have not found documentation for. Where is this version stuff documented?

**Review by Thomas1664 — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**